### PR TITLE
Clean sdk dependency from core

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by D-W- — DISMISSED** [review]

File: src/promptflow/promptflow/core/_utils.py [hunk 1]
```diff
@@ -143,3 +147,92 @@ def init_executable(*, flow_dag: dict = None, flow_path: Path = None, working_di
 
     # for DAG flow, use data to init executable to improve performance
     return ExecutableFlow._from_dict(flow_dag=flow_dag, working_dir=working_dir)
+
+
+# !!! Attention!!!: Please make sure you have contact with PRS team before changing the interface.
```
Comment (wangchao1230) [inline src/promptflow/promptflow/core/_utils.py:152]: will this break PRS when we move this function?

Comment (brynn-code) [reply src/promptflow/promptflow/core/_utils.py:152]: I remember they are using the FlowExecutor.update_environment_variables_with_connections, I'll double confirm about this

File: src/promptflow/promptflow/_cli/_pf/_flow.py [hunk 2]
```diff
@@ -554,20 +554,24 @@ def _resolve_python_flow_additional_includes(source) -> Path:
...
 
 def serve_flow_python(args, source):
+    from promptflow._sdk._configuration import Configuration
     from promptflow.core._serving.app import create_app
 
     static_folder = args.static_folder
     if static_folder:
         static_folder = Path(static_folder).absolute().as_posix()
     config = list_of_dict_to_dict(args.config)
+    pf_config = Configuration(overrides=config)
```
Comment (elliotzh) [inline src/promptflow/promptflow/_cli/_pf/_flow.py:564]: Configuration is still in SDK?

File: src/promptflow/promptflow/_cli/_pf/_flow.py [hunk 3]
```diff
@@ -37,9 +37,9 @@
     copy_extra_files,
 )
 from promptflow._cli._utils import _copy_to_flow, activate_action, confirm, inject_sys_path, list_of_dict_to_dict
-from promptflow._constants import FlowLanguage
+from promptflow._constants import ConnectionProviderConfig, FlowLanguage
 from promptflow._sdk._configuration import Configuration
-from promptflow._sdk._constants import PROMPT_FLOW_DIR_NAME, ConnectionProvider
+from promptflow._sdk._constants import PROMPT_FLOW_DIR_NAME
```
Comment (elliotzh) [inline src/promptflow/promptflow/_cli/_pf/_flow.py:42]: from promptflow._constants import PROMPT_FLOW_DIR_NAME ?

there is another PROMPT_FLOW_DIR_NAME there

File: src/promptflow/promptflow/_cli/_pf/_flow.py (hunk 2 above)
Comment (brynn-code) [reply src/promptflow/promptflow/_cli/_pf/_flow.py:564]: Yes, this is CLI file, it's ok to import SDK

File: src/promptflow/promptflow/_cli/_pf/_flow.py (hunk 3 above)
Comment (brynn-code) [reply src/promptflow/promptflow/_cli/_pf/_flow.py:42]: I'll let sdk constants reference the promptflow constants one

File: src/promptflow/promptflow/core/_utils.py (hunk 1 above)
Comment (brynn-code) [reply src/promptflow/promptflow/core/_utils.py:152]: Just confirmed.

File: src/promptflow/promptflow/_internal/__init__.py [hunk 4]
```diff
@@ -106,6 +102,10 @@
     streaming_response_required,
     validate_request_data,
 )
+from promptflow.core._utils import (
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_internal/__init__.py:105]: there is conflict in this file

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/promptflow/core/_utils.py (hunk 1 above)
Comment (wangchao1230) [reply src/promptflow/promptflow/core/_utils.py:152]: mmm.. then we should remove this attention now

File: src/promptflow/promptflow/core/_connection_provider/_local_connection_provider.py [hunk 5]
```diff
@@ -1,14 +1,19 @@
 # ---------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
+from .._errors import MissingRequiredPackage
 from ._connection_provider import ConnectionProvider
 
 
 class LocalConnectionProvider(ConnectionProvider):
```
Comment (thy09) [inline src/promptflow/promptflow/core/_connection_provider/_local_connection_provider.py:8]: Maybe move to _sdk

**Review by thy09 — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**